### ISSUE TITLE
fp16

### ISSUE THREAD

**Issue by nationalflag** [layumi/Person_reID_baseline_pytorch#95]
pytorch0.4.1可以用fp16训练吗？用fp16训练，测试还是可以用fp32吧

**Comment by layumi** [comment]
Hi @nationalflag ,

You may have a try. I recommend to use the latest stable release pytorch 1.0, obtainable from https://pytorch.org/.

When testing, I also use f16 to extract feature, and save the feature into fp32.

**Comment by nationalflag** [comment]
你说的性能没有下降，是训练就是用的fp16，还是训练用fp32，测试用fp16？

**Comment by layumi** [comment]
I used fp16 to train and test the model.

**Comment by nationalflag** [comment]
感觉如果训练用fp16，batchsize可以设大一些，说不定结果还会更好呢？

**Comment by layumi** [comment]
Sure. You may have a try.

**Comment by nationalflag** [comment]
Thanks~

**Comment by nationalflag** [comment]
你有用fp16训过triple loss吗？不知道为什么，loss直接是nan

**Comment by layumi** [comment]
Hi @nationalflag ,

Just for a reminder,  you also need to change the input to fp16 and change the optimizer to FP16.optimizer.

For reference, you may look the imagenet demo at (https://github.com/NVIDIA/apex/tree/master/examples/imagenet).

**Comment by nationalflag** [comment]
我试过了，只用softmax可以正常训，用triple loss就不行

**Comment by nationalflag** [comment]
是梯度溢出了，我在FP16_Optimizer用dynamic_loss_scale，就一直提示‘OVERFLOW! Skipping step. Attempted loss scale’
